#! /bin/sh /usr/share/dpatch/dpatch-run
## -*- diff -*-
## polyfit-scaling.dpatch by Rafael Laboissiere <rafael@debian.org>
##
## DP: Addapt SparseCholError prototype to SuiteSparse 3.2.0

@DPATCH@
--- octave3.0-3.0.3.orig/liboctave/sparse-util.cc
+++ octave3.0-3.0.3/liboctave/sparse-util.cc
@@ -31,7 +31,7 @@
 #include "sparse-util.h"
 
 void
-SparseCholError (int status, char *file, int line, char *message)
+SparseCholError (int status, const char *file, int line, const char *message)
 {
   (*current_liboctave_warning_handler)("warning %i, at line %i in file %s",
 				     status, line, file);
--- octave3.0-3.0.3.orig/liboctave/sparse-util.h
+++ octave3.0-3.0.3/liboctave/sparse-util.h
@@ -24,7 +24,7 @@
 #if !defined (octave_sparse_util_h)
 #define octave_sparse_util_h 1
 
-extern OCTAVE_API void SparseCholError (int status, char *file, int line, char *message);
+extern OCTAVE_API void SparseCholError (int status, const char *file, int line, const char *message);
 extern OCTAVE_API int SparseCholPrint (const char *fmt, ...);
 
 #endif
